buildSrc/src/main/java/xyz/wagyourtail/doclet/pydoclet/parsers/ClassParser.java

```diff
@@ -0,0 +1,353 @@
+        put("java.lang.Object", "object");
+        put("java.lang.String", "str");
+        put("java.lang.Integer", "int");
+        put("java.lang.Boolean", "bool");
+        put("java.lang.Double", "float");
+        put("java.lang.annotation.Annotation", "");
+        put("java.lang.Enum", "");
+        put("java.util.Iterator", "iter");
+    }},
+                                          withArg = new HashMap<>(){{
+        put("java.util.Set", "List");
+        put("java.util.List", "List");
+        put("java.util.Map", "Mapping");
+        put("java.util.Collection", "List");
+    }},
+                                          unwantedClass = new HashMap<>(){{
+        put("java.lang.Object", "");
+        put("java.lang.annotation.Annotation", "");
+        put("java.lang.Enum", "");
+        put("java.util.Collection", "");
+    }};
+
+
+
+    public ClassParser(TypeElement type) {
+        this.type = type;
+    }
+
+
+    public String parseClass(){
+        StringBuilder sb = new StringBuilder();
+
+        //Classline e.g class Test(AbstractTest, ):
+        sb.append(getClassLine());
+
+        //Fields
+        sb.append(getFields()).append("\n");
+
+        //Constructor
+        sb.append(getConstructor());
+
+        //Methods
+        sb.append(getMethods());
+
+        //safety pass
+        sb.append(getTabs(1) + "pass\n\n");
+
+        //imports + abstract types
+        sb.insert(0, getImports()).append("\n");
+
+        //Main.reporter.print(Diagnostic.Kind.NOTE, type + ", " + imports + "");
+
+        return sb.toString();
+    }
+
+    private String getConstructor() {
+        StringBuilder sb = new StringBuilder();
+
+
+        return sb.toString();
+    }
+
+    private String getMethods() {
+        StringBuilder sb = new StringBuilder();
+
+
+        type.getEnclosedElements().stream().filter(e -> e.getKind() == ElementKind.METHOD || e.getKind() == ElementKind.CONSTRUCTOR).forEach(el ->{
+            if (!el.getModifiers().contains(Modifier.PUBLIC)) return;
+            ExecutableElement method = (ExecutableElement) el;
+            importOverload = true;
+            sb.append(getTabs(1)).append("@overload\n");
+            sb.append(getTabs(1)).append("def ");
+            if(el.getKind() == ElementKind.CONSTRUCTOR){
+                sb.append("__init__");
+            }else{
+                sb.append(method.getSimpleName());
+            }
+
+            sb.append("(self");
+
+            method.getParameters().forEach(parameter -> {
+                sb.append(", ").append(parameter.getSimpleName()).append(": ").append(getTypeMirrorName(parameter.asType(), false));
+            });
+            sb.append(") -> ");
+            //Main.reporter.print(Diagnostic.Kind.NOTE, getTypeMirrorName(method.getReturnType(), false) + "");
+            if (method.getReceiverType() != null) sb.append(getTypeMirrorName(method.getReturnType(), false));
```

Comment:
this is putting self references inside "" when it doesn't need to. (see `CommandBuilder` for that)

Reply:
I don't see any problem in this, because in the [PEP 484](https://www.python.org/dev/peps/pep-0484/#forward-references) self-references and the solution are mentioned with "". Also, note that just setting CommandBuilder instead of "CommandBuilder" does not seem work. But one thing to fix would be the import of .CommandBuilder at the beginning of the file.